packages/komencikit/src/actions.ts

```diff
@@ -0,0 +1,144 @@
+import { RawTransaction } from '@celo/contractkit/lib/wrappers/MetaTransactionWallet'
+import * as t from 'io-ts'
+
+export enum ActionTypes {
+  StartSession = 'StartSession',
+  DeployWallet = 'DeployWallet',
+  DistributedBlindedPepper = 'DistributedBlindedPepper',
+  SubmitMetaTransaction = 'SubmitMetaTransaction',
+  RequestSubsidisedAttestation = 'RequestSubsidisedAttestation',
+}
+
+export interface Action<TAction, TPayload, TResp> {
+  method: string
+  action: TAction
+  path: string
+  payload: TPayload
+  codec: t.Type<TResp>
+}
+
+type ActionFactory<TAction, TPayload, TResp> = (
+  payload: TPayload
+) => Action<TAction, TPayload, TResp>
+
+export const action = <TAction, TPayload, TResp>(
+  action: TAction,
+  method: string,
+  path: string,
+  codec: t.Type<TResp>
+): ActionFactory<TAction, TPayload, TResp> => (payload) => ({
+  method,
+  path: path,
+  action: action,
+  codec,
+  payload: {
+    ...payload,
+  },
+})
+
+export interface StartSessionPayload {
+  captchaResponseToken: string
+  externalAccount: string
+  signature: string
+}
+
+export const StartSessionResp = t.type({
+  token: t.string,
+})
+
+export type StartSessionResp = t.TypeOf<typeof StartSessionResp>
+
+export const startSession = action<ActionTypes.StartSession, StartSessionPayload, StartSessionResp>(
+  ActionTypes.StartSession,
+  'POST',
+  'v1/startSession',
+  StartSessionResp
+)
+
+interface GetDistributedBlindedPepperPayload {
+  e164Number: string
```

Comment:
We should be blinding and unblinding the phone number on the client before it makes its way to Komenci.

Reply:
Sorry just realized we were doing this _within_ Komenci and it needs to be moved out to this client to avoid giving Komenci the phonenumber <> on-chain id mapping. We can do that in a subsequent PR.

Reply:
I actually had this on the back of my mind for a while and wanted to ask you about it, because I also had the impression that we had to blind the PN but then saw that komenci was expecting the real thing. Good to know that we need to blind it 👍.